modules/umig/main.tf

```diff
@@ -56,7 +56,7 @@ resource "google_compute_instance_from_template" "compute_instance" {
     network_ip         = length(var.static_ips) == 0 ? "" : element(local.static_ips, count.index)
 
     dynamic "access_config" {
-      for_each = var.access_config
+      for_each = var.access_config[count.index]
```

Comment:
Looks like our integration tests are failing here. In particular, this [example](https://github.com/terraform-google-modules/terraform-google-vm/blob/master/examples/umig/static_ips/main.tf) when we default `access_config` to `[]`. Perhaps we can have a check to make sure we only access `var.access_config[count.index]` if `length(var.access_config) > 0`? This however would mean if num_instances or static_ips > 1 and we want `access_config` for only one instance, we will need to provide empty lists like 
```hcl
[[{
  // when static_ips=4
    nat_ip       = google_compute_address.ip_address.address
    network_tier = "PREMIUM"
  }],[],[],[]]
```

Reply:
I think this will solve the issue. And also one wouldn't need to provide a list as yours, but just with a list with first element.

```terraform
dynamic "access_config" {
  # convert to map to use lookup function with default value
  for_each = lookup({for k, v in var.access_config: k => v}, count.index, [])
  content {
    nat_ip       = access_config.value.nat_ip
    network_tier = access_config.value.network_tier
  }
}
```

_Lookup with a default value works just with maps and didn't find anything similar for lists._